python/cudf_polars/cudf_polars/dsl/ir.py

```diff
@@ -703,7 +703,12 @@ def __init__(
         self.df = df
         self.projection = tuple(projection) if projection is not None else None
         self.predicate = predicate
-        self._non_child_args = (schema, df, self.projection, predicate)
+        self._non_child_args = (
+            schema,
+            pl.DataFrame._from_pydf(df),
+            self.projection,
+            predicate,
+        )
```

Comment:
Just a note that I pulled these DataFrameScan changes into https://github.com/rapidsai/cudf/pull/17441 (in case that gets in first).

Reply:
Scratch that. Latest version of that PR doesn't need to touch this logic anymore.